feat:  점심추천에 동의한 고객들에게 메세지 전송하는 기능 추가 (Discord Webhook 사용)

HyunsooZo opened the pull request:
### 작업 내용
- 점심추천에 동의한 고객들에게 메세지 전송하는 기능 추가 
- `QueryDSL` 의존성및 설정 추가 

### 변경 사항(추가시엔 추가사항)
- **점심추천 메세지 전송 프로세스**
   1. 점심약속에 동의한 고객들의 목록을 조회
   2. 고객의 좌표에 가까우며 별점이 가장높은 5개 카테고리의 식당 5개씩, 총 25개 조회(`QueryDSL`,하버사인 공식 사용)
   3. 5개씩 한 `embed`에 묶어 `DiscordWebhook` 으로 전송할 메세지 객체 생성
   4. 조회된 고객순서대로 메세지 전송 `DiscordWebhook` 호출
   
- **메시지 예시(각 카테고리별 5개씩)**
   >***오늘의 추천 일식***
**산(뼈찜,뼈곰탕)**
경기도 평택시 탄현로1번길 11, 101,102호 (장당동, 엘림하우스)
**스고이**
경기도 평택시 고덕갈평7길 10, 1층 (고덕동)
**광명회수산**
경기도 평택시 현촌4길 2-33, 101호 (용이동)
**오늘은참치**
경기도 시흥시 옥구천동로 449, 부성파스텔아파트 상가동 1층 105호 (정왕동)
**장군수산**
경기도 오산시 오산로160번길 5-6, 102,103,104호 (원동, 건정프라자)

### 특이 사항
- 해당 동작 로컬에서 테스트 컨트롤러 직접 호출하여 테스트 확인했습니다. 
- `dev`에 적용되지 않은 코드들과 호환 체크하느라 준규님 브랜치(PR머지안된상태) 를 머지해서
   커밋목록이 조금 깁니다. 아래 6개 커밋만 리뷰 부탁드립니다. 

### 테스트
- [x] api 테스트

### 관련 이슈(옵셔널)
- 없음

HyunsooZo committed:
chore: JSON 파싱을 위한 jackson 라이브러리 의존성 추가

HyunsooZo committed:
remove: 프로젝트 내 중복파일 삭제

HyunsooZo committed:
chore: 스케줄링 및 BaseEntity 적용을 위한 어노테이션 추가

HyunsooZo committed:
feat: 식당 정보 가져오는 스케쥴링 기능 추가

총 5개 외부 api를 호출합니다. (패스트푸드,중식,양식,뷔페,일식)
- 모든 값을 그대로 저장하되 null값은 데이터 타입에 따라 "데이터없음" , 0 , 0.0 등 으로 전처리
- 유일키는 식당이름+지번주소 에 공백을 제거하여 사용
- "폐업" 식당의 경우 저장하지 않음.
- 매일 23:59 스케줄링 동작
- 저장시점에 저장 식당 종류, 시간을 로깅

HyunsooZo committed:
feat: 기존 식당 정보 존재 시 업데이트 하는 기능 추가

영업점상태(폐업,영업)
- 폐업시 삭제 영업시 업데이트
Restaurant 필드 추가
- (BaseEntity 업데이트,생성일)
- 평점 , 리뷰 개수

HyunsooZo committed:
style: 가독성을 위한 줄바꿈 및 불필요한 어노테이션 삭제

HyunsooZo committed:
Merge remote-tracking branch 'origin/dev' into dev

junkyu92 committed:
feat: 리뷰 등록 api 추가

junkyu92 committed:
style: 테스트코드 given when then 분리

junkyu92 committed:
refactor: 메서드명 변경, 사용하지 않는 변수 제거

HyunsooZo committed:
Merge remote-tracking branch 'origin/feature/review/create' into dev

HyunsooZo committed:
chore: QueryDsl 의존성 추가

HyunsooZo committed:
chore: QueryDsl로 자동생성되는 디렉토리 gitignore에 추가

HyunsooZo committed:
feat: QueryDsl 레포지토리와의 빈 경합을 방지하기 위해 Primary 어노테이션 추가

HyunsooZo committed:
feat: QueryDSL 설정파일 추가

HyunsooZo committed:
style: 주석 보완 및 불필요한 주석 제거

HyunsooZo committed:
feat: 점심추천에 동의한 고객들에게 Discord 메세지 전송하는 기능 추가

점심추천 메세지 전송 프로세스
1. 점심약속에 동의한 고객들의 목록을 조회
2. 고객의 좌표에 가까우며 별점이 가장높은 5개 카테고리의 식당 5개씩, 총 15개 조회(QueryDSL사용)
3. 5개씩 한 embed에 묶어 DiscordWebhook 으로 전송할 메세지 객체 생성
4. 조회된 고객순서대로 메세지 정송 웹훅 호출

madst0614 reviewed (approved):
코드 깔끔하게 잘 적어주신 것 같습니다! 유저 기능들 공유가 필요할 것 같은데, 내일 정리해서 넘겨드릴게요!

HyunsooZo committed:
feat: 맛집 추가 시 별점 및 리뷰 개수 0(0.0)으로 초기화 하는 코드 추가

HyunsooZo committed:
feat: 점심추천 시 별점 함께 노출하도록 수정

junkyu92 commented on src/main/java/wanted/n/repository/RestaurantQRepositoryImpl.java:
```diff
@@ -0,0 +1,48 @@
+            BUFFET.getValue(),
+            FAST_FOOD.getValue(),
+            CAFE.getValue()
+    );
```
참고만 해주세요.
카테고리 종류가 많아지면 이 부분은 스트림 사용해서 가져올 수 있을 것 같습니다.
Arrays.stream(RestaurantCategory.values()).map(RestaurantCategory::getValue).collect(Collectors.toList());

junkyu92 reviewed (approved):
전체적으로 깔끔하게 잘 작성하신 것 같습니다. 
고생하셨습니다.

HyunsooZo committed:
refactor: 식당 카테고리의 유연성과 확장성 확보를 위한 스트림 코드 적용

RestaurantCategory 값에 대한 getValue() 메서드 호출을 수동으로 하나씩 나열하던코드를

Arrays.stream()으로 배열을 스트림으로 변환하고,
map()을 사용하여 RestaurantCategory 열거형의 각 요소에 getValue() 메서드를 적용하고,
collect() 메서드를 사용하여 결과를 리스트로 수집

12hyeon reviewed (approved):
깔끔하게 정리된 메시지 전송 로직 확인했습니다!

junkyu92 committed:
Merge branch 'dev' into feature/recommendation
